modules/core/src/directx.cpp

```diff
@@ -1397,6 +1397,33 @@ void convertFromD3D11Texture2D(ID3D11Texture2D* pD3D11Texture2D, OutputArray dst
 #endif
 }
 
+#ifdef HAVE_OPENCL
+void convertFromD3D11Texture2DtoCLImage(ID3D11Texture2D* pD3D11Texture2D, cl_mem* clImage)
+{
+    CV_UNUSED(pD3D11Texture2D); CV_UNUSED(clImage);
+#if !defined(HAVE_DIRECTX)
+    NO_DIRECTX_SUPPORT_ERROR;
+#elif !defined(HAVE_OPENCL)
+    NO_OPENCL_SUPPORT_ERROR;
+#else
+    ID3D11Device *ppDevice = nullptr;
+    pD3D11Texture2D->GetDevice(&ppDevice);
+    cv::ocl::Context ctx = ocl::initializeContextFromD3D11Device(ppDevice);
```

Comment:
in general it is incorrect: 
initialize context - means creation NEW context, which may be differ from context used for texture creation. Need to extract Context from Device (or texture preferably)  then cast it to OPENCL context using KHR

Reply:
Changed. But approach the same